src/Components/SharedComponents/PlantPhoto/PlantPhoto.js

```diff
@@ -1,12 +1,118 @@
 import React from "react";
+import axios from 'axios';
+import AddNewPlantPhoto from "./AddNewPlantPhoto/AddNewPlantPhoto";
+import Snackbar from "@material-ui/core/Snackbar";
+import Alert from '../../SharedComponents/Alert/Alert';
+import DeletePlantPhotoButton from "./DeletePlantPhotoButton/DeletePlantPhotoButton";
 import './plantPhoto.scss';
 
-function PlantPhoto(props){
-    return(
-        <div className={`plantPhoto__container ${ props.fullDescriptionView && 'largePhoto'}`}>
-            <img className="plantPhoto" src="/plantsPhotos/monstera.png" alt="" />
-        </div>
-    )
+class PlantPhoto extends React.Component {
+    state = {
+        plant: this.props.plant,
+        open: false
+    }
+
+    componentDidMount() {
+        this.state.plant.imageUrl && this.fetchPlantImg()
+    }
+
+    handleClick = () => {
+        this.setState({
+            open: true
+        });
+    };
+
+    handleClose = (event, reason) => {
+        if (reason === 'clickaway') {
+            return;
+        }
+        this.setState({
+            open: false
+        });
+    };
+
+    fetchPlantImg = () => {
+        let url = this.state.plant.imageUrl || `plants/${this.state.plant.id}/image`
+
+        axios.get(url,{ responseType: "arraybuffer" })
+            .then((response) => {
+                const data = `data:${response.headers['content-type']};base64,${new Buffer(response.data, "binary").toString('base64')}`;
+                this.setState({imgSrc: data});
+            })
+            .catch(err => {
+                console.log(err)
+            })
+    }
+
+    uploadPlantImage = (photo) => {
+        const data = new FormData();
+        data.append('image', photo);
+
+        (async () => {
+            axios.interceptors.response.use(
```

Comment:
interceptors should be configured near root level. As we agreed move this logic inside catch of post

Reply:
Fixed